src/GameEngine/GameEngineDriver.cpp

```diff
@@ -149,5 +149,40 @@ namespace GameEngine {
             return true;
         }
 
+        bool test_tournament(){
+            Terminal::test_mode_on();
+            auto tournament_1 = new Tournament();
+            vector<int> input_vector = {2, 2, 7, 12, 2, 3, 3, 12};
+            Terminal::set_input_vector(input_vector);
+
+            // (2) when the tournament mode is chosen, the user is asked to select
+            tournament_1->prepareTournament(); // asks for num_maps, num_players, num_games, num_turns
+            Terminal::clear_terminal_input_counter();
+            // (3) after being started, the tournament runs without any user interaction
+            tournament_1->start();
+            // (4) upon completion, the results of the tournament are displayed as depicted above
+            tournament_1->displayResults();
+            //When finished, deleting everything
+            GameEngine::instance()->game_state->clear();
+
+            //assert( = );
+            GameEngine::instance()->reset_test();
+            return true;
+        }
+
+        void tournament_driver(){
+            //TO BE MOVED IF NECESSARY
+            // (2) when the tournament mode is chosen, the user is asked to select
+                auto tournament_1 = new Tournament();
+                tournament_1->prepareTournament(); // asks for num_maps, num_players, num_games, num_turns
+            // (3) after being started, the tournament runs without any user interaction
+                tournament_1->start();
+            // (4) upon completion, the results of the tournament are displayed as depicted above
+                tournament_1->displayResults();
+                //When finished, deleting everything
+                GameEngine::instance()->game_state->clear();
+                tournament_1->~Tournament();
```

Comment:
Don't call the deconstructor. Call delete.

Reply:
Test not testing anything too

Reply:
@derkanadier

Reply:
Well at the end there is
GameEngine::instance()->game_state->clear();
GameEngine::instance()->reset_test();

I found that if I add delete tournament_1; before those two lines, the program executes till the end but then crashes. However, If I write the delete command, keep the game_state->clear() and ommit the rest_test then the program executes and exits as intended.